Puzzlepart/prosjektportalen365
# Dynamic and improved aggregated lists (Portfolio)

### Your checklist for this pull request


### Description

A big change to the way the aggregated webparts at the Portfolio level works. Implementation of the same functionality found in the PortfolioOverview webpart. This includes Views, filtering, grouping, etc...

Aggregated webparts included in Prosjektportalen 365 that will be affected:
- Gevinstoversikt
- Erfaringslogg
- Leveranseoversikt
- Risikooversikt

**NOTE!**
The improvements made here does not affect Program/Overordnet. The aggregated webparts that came with 'Program- and Overordnet-område' are created as unique webparts instead of using the Portfolio aggregated webpart directly and use the 'Datakilder' list to set up the source.

Because of this we will have to keep Program and Project at the pp365-portfoliowebpart@1.5.4-679.2 version until we have created new webparts for Program/Overordnet that will work directly and more seamlessly with the new additions made here.

## Additions:
- Added button to 'Konfigurasjon av Prosjektportalen' page.
- Added a new list.

## Provision template additions/changes:
- New List (*Title, Url added to resource files*):
  - Lists/Prosjektinnholdskolonner

- New SiteFields (*Name, Descriptions added to resource files.*)
  - GtODataQuery (Note field to filter the projects found within the datasource searchquery)
  - GtProjectContentColumns (Lookup to 'Prosjektinnholdskolonner')
  - GtProjectContentGroupBy (Lookup to 'Prosjektinnholdskolonner')
  - GtProjectContentRefiners (Lookup to 'Prosjektinnholdskolonner')

- Changes to Datakilder
  - Added the new SiteFields to the list
  - Reorder of columns in list

- Changes to datakilde contenttype
  - Added the new SiteFields to the contenttype

- Changes to the following ClientSidePages
  - Erfaringslogg, Leveranseoversikt, Risikooversikt, Gevinstoversikt
    - Removed all data for columns property in the webpart
    - Set Overwrite to true
  - Gevinstoversikt
    - Switched out the webpart to use aggregated webpart instead of the benefits webpart
  - Konfigurasjon
    - Added button (QuickLink) for 'Prosjektinnholdskolonner' list to 'Konfigurasjon av Prosjektportalen' page.

## Changes

**Aggregated webpart**

- Views
  - Ability to create new views
  - Possibility to see the views from 'Datakilder' in the drop-down menu with 'Edit' functionality.
![image](https://user-images.githubusercontent.com/28678468/172369484-62e4fab7-4d70-4453-bd90-149783e71663.png)
  
- Filtering
  - Ability to filter the view based on the selected columns from 'Datakilder'
![image](https://user-images.githubusercontent.com/28678468/172369529-52fdeb37-6219-45ad-bb2d-f33f42e5d54b.png)

- Group by
  - Ability to set a pre-defined group by column based on the selected column from 'Datakilder'

- Change in 'Datakilder' list (4 new columns)
![image](https://user-images.githubusercontent.com/28678468/172369641-71b0ba01-fde0-49e5-8223-81d3daec96a5.png)

- Columns
  - Columns defined in 'Datakilder' will be used instead of the columns found in webpart properties
  - Columns are editable same as before where all data is persisted

- Edit mode
    - 'Legg til kolonne':
      - This will now show a dropdown instead of opening the Column panel (Legg til kolonne/Vis/skjul kolonner)
      - When creating a new column, the column is also created in the 'Prosjektinnholdskolonner' list and connected to the 'Datakilde' element for the current view
      - Deleting a column will also remove this from the 'prosjektinnholdskolonner' list
        - Columns: 'Prosjekt' and 'Tittel' can't be deleted.
      - Editing the name for the columns in an aggregated list will only persist the changes directly in the list and won't affect the element in the 'Prosjektinnholdskolonner' list
    - Added new option: 'Vis/skjul kolonner', ability to remove columns from Datakilder for the current view
      - Tried to replicate this to be identical to SharePoint out of the box version

### How to test

Test out the changes made the aggregated webpart

1. Check to see if 'Prosjektinnholdskolonner' and 'Datakilder' is correctly set up after upgrade
2. Test creating, changing and editing views
3. Test filtering
4. Test group by
5. Edit columns and test deleting custom columns
6. Test 'vis/skjul kolonner'
7. Make changes to 'Datakilder' list
   a. Test odata query

### Relevant issues (if applicable)

#706
